src/test/java/com/icthh/xm/ms/balance/service/BalanceServiceUnitTest.java

```diff
@@ -1825,4 +1834,499 @@ public void setIsLastBalanceChangeEvent() {
+        revertBalanceOperationRequest.setEndDateTime(endDateTime);
+        revertBalanceOperationRequest.setLabel(label);
+        when(balanceChangeEventRepository.findBalanceChangeEventsByOperationId(revertBalanceOperationRequest.getUuid()))
+            .thenReturn(emptyList());
+
+        BalanceChangeEvent existBalanceChangeEvent = createBalanceEvent("90", balanceId, CHARGING, "10", "100");
+        existBalanceChangeEvent.setOperationId(uuidToRevert);
+        existBalanceChangeEvent.setPocketChangeEvents(List.of(
+            createPocketEvent("90", pocketId, "key", "label", "10", "100")
+        ));
+        when(balanceChangeEventRepository.findBalanceChangeEventsByOperationId(uuidToRevert))
+            .thenReturn(List.of(existBalanceChangeEvent));
+        when(balanceChangeEventRepository.existsByRevertOperationId(uuidToRevert))
+            .thenReturn(false);
+
+        createBalance(balanceId);
+        expectedAuth();
+        when(balanceRepository.findBalanceAmount(any(), any())).thenReturn(of(new BigDecimal(10)));
+
+        Pocket pocket = new Pocket().key("ASSERTION_KEY").label("label").amount(new BigDecimal("90"));
+        pocket.setId(pocketId);
+        when(pocketRepository.save(any(Pocket.class))).thenReturn(pocket);
+
+        buildExpectedBalanceChangeEvent(
+            new BigDecimal(90), REVERT_CHARGING, currentUuid, new BigDecimal(10),
+            new BigDecimal(100), balanceId, new BigDecimal(90), uuidToRevert);
+
+        BalanceChangeEventDto result = balanceService.revertBalanceOperation(revertBalanceOperationRequest);
+
+        verify(balanceChangeEventRepository).save(
+            argThat(balanceChangeEvent -> balanceChangeEvent.getAmountDelta().compareTo(result.getAmountDelta()) == 0
+                && balanceChangeEvent.getOperationType().equals(result.getOperationType())
+                && balanceChangeEvent.getOperationId().equals(result.getOperationId())
+                && balanceChangeEvent.getAmountBefore().compareTo(result.getAmountBefore()) == 0
+                && balanceChangeEvent.getAmountAfter().compareTo(result.getAmountAfter()) == 0
+                && balanceChangeEvent.getBalanceId().equals(result.getBalanceId())
+                && balanceChangeEvent.getAmountTotal().compareTo(result.getAmountTotal()) == 0
+                && balanceChangeEvent.getRevertOperationId().equals(result.getRevertOperationId()))
+        );
+        verify(balanceRepository).findOneByIdForUpdate(balanceId);
+        verify(balanceChangeEventRepository).findBalanceChangeEventsByOperationId(currentUuid);
+        verify(balanceChangeEventRepository).findBalanceChangeEventsByOperationId(uuidToRevert);
+        verify(balanceChangeEventRepository).existsByRevertOperationId(uuidToRevert);
+        verify(balanceChangeEventRepository).findLastBalanceChangeEvent(balanceId);
+        verify(balanceRepository).findBalanceAmount(any(), any());
+        verify(pocketRepository).save(any(Pocket.class));
```

Comment:
pls replace any to assertion that call was correct

Reply:
Fixed